app/services/invite_provider_user.rb

```diff
@@ -0,0 +1,60 @@
+require 'jwt'
+
+class InviteProviderUser
+  def initialize(provider_user_form:)
+    @provider_user_form = provider_user_form
+  end
+
+  def call
+    ActiveRecord::Base.transaction do
+      if @provider_user_form.save
+        response = invite_user_to_dfe_sign_in
+        raise DfeSignInApiError.new(response) if response && !response.status.success?
+
+        # TODO: send welcome/invitation email
+        true
+      end
+    end
+  rescue DfeSignInApiError => e
+    e.errors.each { |error| @provider_user_form.errors.add(:base, error) }
+    false
+  end
+
+  def dfe_invite_url
+    baseurl = ENV['DSI_API_URL']
+    if baseurl.present?
+      "#{ENV['DSI_API_URL']}/services/apply/invitations"
+    end
+  end
+
+  def invite_user_to_dfe_sign_in
+    return unless dfe_invite_url && @provider_user_form.persisted?
```

Comment:
I don't think we have to check for `persisted?` because it'll always be true?

Reply:
Yes, that is correct. I've addressed this.